packages/tools/src/utilities/stackPrefetch/stackContextPrefetch.ts

```diff
@@ -0,0 +1,386 @@
+    imageLoader.loadAndCacheImage(imageId, options);
+
+  const { useNorm16Texture } = getCoreConfiguration().rendering;
+
+  indicesToRequestCopy.forEach((imageIdIndex) => {
+    const imageId = stack.imageIds[imageIdIndex];
+    // IMPORTANT: Request type should be passed if not the 'interaction'
+    // highest priority will be used for the request type in the imageRetrievalPool
+    const options = {
+      targetBuffer: {
+        type: useNorm16Texture ? undefined : 'Float32Array',
+      },
+      preScale: {
+        enabled: true,
+      },
+      requestType,
+    };
+
+    imageLoadPoolManager.addRequest(
+      requestFn.bind(null, imageId, options),
+      requestType,
+      // Additional details
+      {
+        imageId,
+      },
+      priority
+      // addToBeginning
+    );
+  });
+}
+
+function getPromiseRemovedHandler(element) {
+  return function (e) {
+    const eventData = e.detail;
+
+    // When an imagePromise has been pushed out of the cache, re-add its index
+    // It to the indicesToRequest list so that it will be retrieved later if the
+    // CurrentImageIdIndex is changed to an image nearby
+    let stackData;
+
+    try {
+      // It will throw an exception in some cases (eg: thumbnails)
+      stackData = getStackData(element);
+    } catch (error) {
+      return;
+    }
+
+    if (!stackData || !stackData.imageIds || stackData.imageIds.length === 0) {
+      return;
+    }
+
+    const stack = stackData;
+    const imageIdIndex = stack.imageIds.indexOf(eventData.imageId);
+
+    // Make sure the image that was removed is actually in this stack
+    // Before adding it to the indicesToRequest array
+    if (imageIdIndex < 0) {
+      return;
+    }
+
+    const stackPrefetchData = getToolState(element);
+
+    if (
+      !stackPrefetchData ||
+      !stackPrefetchData.data ||
+      !stackPrefetchData.data.length
+    ) {
+      return;
+    }
+
+    stackPrefetchData.indicesToRequest.push(imageIdIndex);
+  };
+}
+
+function onImageUpdated(e) {
+  // Start prefetching again (after a delay)
+  // When the user has scrolled to a new image
+  clearTimeout(resetPrefetchTimeout);
+  resetPrefetchTimeout = setTimeout(function () {
+    const element = e.target;
+
+    // If playClip is enabled and the user loads a different series in the viewport
+    // An exception will be thrown because the element will not be enabled anymore
+    try {
+      updateToolState(element);
+      prefetch(element);
+    } catch (error) {
+      return;
+    }
+  }, resetPrefetchDelay);
+}
+
+// Not a full signum, but good enough for direction.
+const signum = (x) => (x < 0 ? -1 : 1);
+
+const updateToolState = (element) => {
+  const stack = getStackData(element);
+  if (!stack || !stack.imageIds || stack.imageIds.length === 0) {
+    console.warn('CornerstoneTools.stackPrefetch: No images in stack.');
+    return;
+  }
+
+  const { currentImageIdIndex } = stack;
+  const { maxAfter = 5, minBefore = 1 } = configuration;
```

Comment:
there is a default in the top of the page, so I don't think we need to default again

Reply:
Configuration can be supplied not containing the defaults, and I like the actual default configuration to have it, so both are required.